src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -230,6 +237,14 @@ private void WriteMethods(TypeBuilderImpl typeBuilder, List<GenericTypeParameter
             }
         }
 
+        private static int AddMethodBody(MethodBuilderImpl method, ILGeneratorImpl il, MethodBodyStreamEncoder methodBodyEncoder) =>
+            methodBodyEncoder.AddMethodBody(
+                instructionEncoder: il.Instructions,
+                maxStack: 8, // TODO
```

Comment:
Per offline discussion, we should add an implementation to calculate this with this PR.

Reply:
If we can set `maxstack == the number of opcodes` for now that should protect us against any `InvalidProgramException` thrown by the JIT etc.

Reply:
I can do that until I have got a better handling